src/components/field/field.component.jsx

```diff
@@ -0,0 +1,46 @@
+import "./field.style.css";
+import FieldBlock from '../field-block'
+
+
+const Field = ( { field, fieldWidth, fieldHeight, size, rightClickHandler,
+     leftClickHandler, imageRatio, screenRatio,  back, med_box }) => {
+
+    const blocks = [];
+ 
+    for (let y = 0; y < field.length; y++) {
+
+       
+        for (let x = 0; x < field[0].length; x++) {
+
+            if (field[y][x].val !==-2) {
+                blocks.push(<FieldBlock key = {`${y}-${x}`}
+                block = { field[y][x] }
+                imageRatio = { imageRatio }
+                back = { back}
+                med_box = { med_box }
+                screenRatio = { screenRatio }
+                rightClickHandler = { rightClickHandler }
+                leftClickHandler = { leftClickHandler } />);  
+                
+            }

```

Comment:
I'm sure that it possible to rewrite it without nested loops and mutable `push`, but with either `map` or `reduce` instead. It's more convenient and makes code clearer